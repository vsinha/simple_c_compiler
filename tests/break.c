
long i;

void main()
{
	for (i=0; i<15; i= i+1) {
		if (i==5) {
			break;
		}
		printf("i=%d\n", i);
	}
	printf("for i=%d\n", i);

	i = 0;
	while ( i<15) {
		printf("i=%d\n", i);

		if (i==8) {
			break;
		}
		
		i = i+1;
	}
	printf("while i=%d\n", i);
	
	i = 0;
	do {
		printf("i=%d\n", i);

		if (i==10) {
			break;
		}
		
		i = i+1;
	}
	while (i<15);
	
	printf("do/while i=%d\n", i);
	
	printf("OK\n");
}

