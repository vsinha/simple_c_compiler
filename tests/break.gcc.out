break.c: In function 'main':
break.c:10:3: warning: incompatible implicit declaration of built-in function 'printf' [enabled by default]
break.c:10:3: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
break.c:12:2: warning: incompatible implicit declaration of built-in function 'printf' [enabled by default]
break.c:12:2: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
break.c:16:3: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
break.c:24:2: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
break.c:28:3: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
break.c:38:2: warning: format '%d' expects argument of type 'int', but argument 2 has type 'long int' [-Wformat]
